### PR TITLE
cache tokens on username

### PR DESCRIPTION
**Pull request by rickardenglund**
Allow caching of logins with multiple users

### CODE REVIEW

File: v2/auth/cachedauth/sign_in.go [hunk 1]
```diff
@@ -50,16 +62,16 @@ func SignIn(ctx context.Context, username, password string) (err error) {
...
 		return nil
 	}
 
-	tokens, err = auth.SignIn(ctx, username, password)
-	if err != nil {
-		tokens = auth.Tokens{}
-		return err
-	}
+	newtokens, err := auth.SignIn(ctx, username, password)
+	tokens[username] = newtokens
```
Comment (Ekberg) [inline v2/auth/cachedauth/sign_in.go:73]: Will newtokens be empty on error? The old code did not assume this?

File: v2/auth/cachedauth/sign_in.go [hunk 2]
```diff
@@ -11,9 +11,11 @@ import (
 )
 
 var lock sync.RWMutex
-var tokens auth.Tokens
 
 var config *Config
+var tokens map[string]auth.Tokens
+
+const latest = "latest"
```
Comment (Ekberg) [inline v2/auth/cachedauth/sign_in.go:18]: There will be no collisions with valid usernames since all usernames are emails?

Comment (rickardenglund) [reply v2/auth/cachedauth/sign_in.go:18]: exactly

File: v2/auth/cachedauth/sign_in.go (hunk 1 above)
Comment (rickardenglund) [reply v2/auth/cachedauth/sign_in.go:73]: Yes, that was the case in the old code aswell it is stated explicitly on line 60 in the old code

**Review by Ekberg — APPROVED** [review]

**Pull request merged**